via/via_system_macos.cpp

```diff
@@ -0,0 +1,1410 @@
+
+                // Parse the JSON file
+                std::ifstream *stream = NULL;
+                stream = new std::ifstream(cur_layer, std::ifstream::in);
+                if (nullptr == stream || stream->fail()) {
+                    PrintBeginTableRow();
+                    PrintTableElement("");
+                    PrintTableElement(generic_string, VIA_ALIGN_RIGHT);
+                    PrintTableElement(cur_ent->d_name);
+                    PrintTableElement("ERROR reading JSON file!");
+                    PrintEndTableRow();
+                } else {
+                    Json::Value root = Json::nullValue;
+                    Json::Reader reader;
+                    if (!reader.parse(*stream, root, false) || root.isNull()) {
+                        // Report to the user the failure and their
+                        // locations in the document.
+                        PrintBeginTableRow();
+                        PrintTableElement("");
+                        PrintTableElement(generic_string, VIA_ALIGN_RIGHT);
+                        PrintTableElement(cur_ent->d_name);
+                        PrintTableElement(reader.getFormattedErrorMessages());
+                        PrintEndTableRow();
+                    } else {
+                        PrintBeginTableRow();
+                        PrintTableElement("");
+                        PrintTableElement(generic_string, VIA_ALIGN_RIGHT);
+                        PrintTableElement(cur_ent->d_name);
+                        PrintTableElement("");
+                        PrintEndTableRow();
+
+                        // Dump out the standard explicit layer information.
+                        GenerateExplicitLayerJsonInfo(cur_layer.c_str(), root);
+                    }
+
+                    stream->close();
+                    delete stream;
+                    stream = NULL;
+                }
+            }
+        }
+        if (!found_json) {
+            PrintBeginTableRow();
+            PrintTableElement("");
+            PrintTableElement(id, VIA_ALIGN_RIGHT);
+            PrintTableElement(folder_loc);
+            PrintTableElement("No JSON files found");
+            PrintEndTableRow();
+        }
+        closedir(layer_dir);
+    } else {
+        PrintBeginTableRow();
+        PrintTableElement("");
+        PrintTableElement(id, VIA_ALIGN_RIGHT);
+        PrintTableElement(folder_loc);
+        PrintTableElement("No such folder");
+        PrintEndTableRow();
+    }
+
+    return res;
+}
+
+ViaSystem::ViaResults ViaSystemMacOS::PrintSystemSdkInfo() {
+    ViaResults result = VIA_SUCCESSFUL;
+    bool sdk_exists = false;
+    std::string sdk_path;
+    std::string sdk_env_name;
+    const char vulkan_so_prefix[] = "libvulkan.dylib.";
+    DIR *sdk_dir;
+    dirent *cur_ent;
+    char *env_value;
+
+    PrintBeginTable("LunarG Vulkan SDKs", 4);
+
+    // First, try environmental variables
+    for (uint32_t dir = 0; dir < 2; dir++) {
```

Comment:
You don't need this loop anymore since there is only one var to check.  As currently written, there is a mismatch between possible values of `dir` (0 and 1) and case statements (0 only).  VIA terminates early when run without the `VULKAN_SDK` env var set because `dir` gets to `1` and ends up in the `default:` error case